lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -554,6 +557,18 @@ newDBLayer trace fp timeInterpreter = do
                 let cpt = CertificatePublicationTime {slotNo, slotInternalIndex}
                 pure (cpt, cert)
 
+        putHeader point =
+                let record = mkBlockHeader point
+                    key = TH.blockHeight record
+                in repsert (BlockHeaderKey key) record
+
+        listHeaders k = do
+            reverse . fmap (fromBlockHeaders . entityVal) <$> selectList [ ]
+                [ Desc BlockHeight
+                , LimitTo k
+                ]
```

Comment:
**Request:** add a very simple property test for `listHeaders` that relates it to the `putHeader` and `dropHeaders` functions.

**Justification:**

By adding a property, we can check that both the SQLite implementation and the model implementation behave in the same way (especially if someone else changes this code in future).

In particular, it seems that the ordering is important here. We'd presumably like to know that both the model and the SQLite implementation return the most-recently-added items first.

(We'd presumably also like to know that both implementations respect the `k` parameter.)

For inspiration, see the existing properties in `lib/core/test/unit/Cardano/Pool/DB/Properties.hs`.

Reply:
> In particular, it seems that the ordering is important here. We'd presumably like to know that both the model and the SQLite implementation return the most-recently-added items first.

They return the oldest item first... it's a bit confusing, but the intersection finding function expects them that way.

Reply:
This is the test:

```hs
prop_putHeaderListHeader
    :: DBLayer IO
    -> [BlockHeader]
    -> Positive Int
    -> Property
prop_putHeaderListHeader DBLayer{..} headers (Positive k) =
    monadicIO (setup >> prop)
  where
    sortedHeaders = sortBy (\a b -> compare (blockHeight a) (blockHeight b))
        . L.nubBy (\a b -> (blockHeight a) == (blockHeight b)) . nub $ headers
    expected = reverse . (if k == 0 then id else take k) . reverse $ sortedHeaders
    setup = run $ atomically cleanDB
    prop = do
        void $ run $ atomically $ forM sortedHeaders putHeader
        lHeaders <- run $ atomically $ listHeaders k
        monitor $ counterexample $ unlines
            [ "Read from DB: " <> show lHeaders
            , "Expected: " <> show expected
            , "Input: " <> show sortedHeaders
            , "k: " <> show k]
        monitor $ classify (null lHeaders) "Empty distributions"
        assert (lHeaders == expected)
        assert (strictlyIncreasing . fmap blockHeight $ lHeaders)

    strictlyIncreasing :: Ord a => [a] -> Bool
    strictlyIncreasing [] = True
    strictlyIncreasing [_] = True
    strictlyIncreasing (x:y:xs) = (x < y) && strictlyIncreasing (y:xs)
```